src/azure-cli/azure/cli/command_modules/appconfig/feature.py

```diff
@@ -1033,26 +989,119 @@ def list_filter(cmd,
 # Helper functions #
 
 
-def __clear_filter(azconfig_client,
-                   feature,
-                   label=None,
-                   yes=False):
+def __list_features(
+    cmd,
+    feature=None,
+    key=None,
+    name=None,
+    label=None,
+    fields=None,
+    connection_string=None,
+    top=None,
+    all_=False,
+    auth_mode="key",
+    endpoint=None,
+    correlationRequestId=None,
+):
+    if key and feature:
+        logger.warning(
+            "Since both `--key` and `--feature` are provided, `--feature` argument will be ignored."
+        )
+
+    if key is not None:
+        key_filter = key
+    elif feature is not None:
+        key_filter = FeatureFlagConstants.FEATURE_FLAG_PREFIX + feature
+    else:
+        key_filter = (
+            FeatureFlagConstants.FEATURE_FLAG_PREFIX + SearchFilterOptions.ANY_KEY
+        )
+
+    azconfig_client = get_appconfig_data_client(
+        cmd, name, connection_string, auth_mode, endpoint
+    )
+    try:
+        retrieved_keyvalues = __list_all_keyvalues(
+            azconfig_client,
+            key_filter=key_filter,
+            label=label if label else SearchFilterOptions.ANY_LABEL,
+            correlationRequestId=correlationRequestId,
+        )
+        retrieved_featureflags = []
+
+        invalid_ffs = 0
+        for kv in retrieved_keyvalues:
+            try:
+                retrieved_featureflags.append(
+                    map_keyvalue_to_featureflag(keyvalue=kv, show_conditions=True)
+                )
+            except ValueError as exception:
+                logger.warning("%s\n", exception)
+                invalid_ffs += 1
+                continue
+
+        if invalid_ffs > 0:
+            logger.warning(
+                "Found %s invalid feature flags. These feature flags will be skipped.",
+                invalid_ffs,
+            )
+
+        filtered_featureflags = []
+        count = 0
+
+        if all_:
+            top = len(retrieved_featureflags)
+        elif top is None:
+            top = 100
+
+        for featureflag in retrieved_featureflags:
+            if fields:
+                partial_featureflags = {}
+                for field in fields:
+                    # featureflag is guaranteed to have all the fields because
+                    # we validate this in map_keyvalue_to_featureflag()
+                    # So this line will never throw AttributeError
+                    partial_featureflags[field.name.lower()] = getattr(
+                        featureflag, field.name.lower()
+                    )
+                filtered_featureflags.append(partial_featureflags)
+            else:
+                filtered_featureflags.append(featureflag)
+            count += 1
+            if count >= top:
+                break
+        return filtered_featureflags
+
+    except Exception as exception:
+        raise CLIError(str(exception))
+
+
+def __clear_filter(azconfig_client, feature, label=None, yes=False):
```

Comment:
Some commands like this make multiple calls. I think it would be good to also pass correlationRequestId to the underlying calls.

Reply:
Updated to add correlation request id to operations in the same activity